aspnetcore/blazor/common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Data/DbContextFactory.cs

```diff
@@ -0,0 +1,43 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+
+namespace BlazorServerDbContextExample.Data
+{
+    /// <summary>
+    /// Factory to generate instances of <see cref="TContext"/>.
+    /// </summary>
+    /// <typeparam name="TContext">The <see cref="DbContext"/> type to provide.</typeparam>
+    public class DbContextFactory<TContext> : IDbContextFactory<TContext> where TContext : DbContext
+    {
+        /// <summary>
+        /// Service provider.
+        /// </summary>
+        private readonly IServiceProvider _provider;
+
+        /// <summary>
+        /// Inject the service provider.
+        /// </summary>
+        /// <param name="provider">The <see cref="IServiceProvider"/> instance.</param>
+        public DbContextFactory(IServiceProvider provider)
```

Comment:
This should create its own SF scope internally and dispose it at the end? Otherwise the IServiceProvider will hold a reference to each created DbContext until it goes away

Reply:
I'm not creating the DbContext through the service provider, that is just used to resolve the context options. It is using ActivatorUtilities which returns a new copy that isn't tracked by the service provider, as far as I understand.

Reply:
@javiercn to test this I changed the factory to hold a collection of `WeakReference` to the db context. I then generated 50+ contexts and forced garbage collection and they all were deallocated. I also ran profiler and I do not show any lingering references. I can share the code if you like, just looking at what is remaining to finish this. Thanks!